lib/erasure/erasure.go

```diff
@@ -99,3 +103,24 @@ func Reconstruct(nValidators uint, chunks [][]byte) ([]byte, error) {
 
 	return res, nil
 }
+
+func ChunksToTrie(chunks [][]byte) (*trie.Trie, error) {
+	chunkTrie := trie.NewEmptyTrie()
+	for i, chunk := range chunks {
+		encodedI, err := scale.Marshal(uint32(i))
```

Comment:
This specifically needs to be a uint32?

Reply:
for i = 6, 
`scale.Marshal(i)` returns [24] but,
`scale.Marshal(uint32(i))` returns [6, 0, 0, 0] which was the same encoding as the Polkadot code.

so, I think it should be uint32.